de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/ScriptHandler.java

```diff
@@ -377,36 +381,43 @@ private void saveGeneratedResources(FskPortObject fskPortObject, File workingDir
     try {
       File newResourcesDirectory = FileUtil.createTempDir("generatedResources");
 
-      try {
-        if (!command.equals("c()") && !command.equals("print([])")) {
+
+      if (!command.equals("c()") && !command.equals("print([])")) {
+        try {
           String[] filenames = runScript(command, exec, true);
 
           // Copy every resource from the working directory
           for (String filename : filenames) {
-            // file parameters can come from another generatedResourceDirectory
-            // from another model; in that case we use that path since it is
-            // not present in the current workingDirectory
-            Path sourceDir = workingDirectory.toPath();
-            Path source = Paths.get(filename.trim());
-            // if a file variable has a path already attached, use that instead
-            File sourceFile = sourceDir.resolve(source).toFile();
-            File targetFile = new File(newResourcesDirectory, source.getFileName().toString());
-            FileUtil.copy(sourceFile, targetFile, exec);
+            try {
+              // file parameters can come from another generatedResourceDirectory
+              // from another model; in that case we use that path since it is
+              // not present in the current workingDirectory
+              Path sourceDir = workingDirectory.toPath();
+              Path source = Paths.get(filename.trim());
+              // if a file variable has a path already attached, use that instead
+              File sourceFile = sourceDir.resolve(source).toFile();
+              File targetFile = new File(newResourcesDirectory, source.getFileName().toString());
+              FileUtil.copy(sourceFile, targetFile, exec);
+            } catch (CanceledExecutionException | IOException e) {
+              throw new ResourceFileNotFoundException(filename);
+            }
           }
+        } catch (REXPMismatchException | IOException e) {
+          throw new VariableNotGlobalException(command,
+              SwaggerUtil.getModelId(fskPortObject.modelMetadata));
         }
-      } catch (Exception e) {
-        e.printStackTrace();
       }
 
       // Save JSON file
-      if(RunnerNodeModel.SAVETOJSON) {
+      if (RunnerNodeModel.SAVETOJSON) {
         File sourceFile = new File(workingDirectory, JsonHandler.JSON_FILE_NAME);
         File targetFile = new File(newResourcesDirectory, JsonHandler.JSON_FILE_NAME);
         FileUtil.copy(sourceFile, targetFile, exec);
       }
-        fskPortObject.setGeneratedResourcesDirectory(newResourcesDirectory);
-    } catch (Exception e) {
-      e.printStackTrace();
+      fskPortObject.setGeneratedResourcesDirectory(newResourcesDirectory);
+
+    } catch (InterruptedException | RException | CanceledExecutionException | IOException e) {

```

Comment:
InterruptedException is not needed here.

Reply:
ok